### PR TITLE
Added more use case when middleware is not a function

### PR DESCRIPTION
**Pull request by yanickrochon**
I noticed, this morning that if `app.use()` is called, or `null` or `undefined` is passed, then `debug` will throw and will not return a descriptive error to the user; it should throw `"middleware must be a function!"` whatever how `app.use` is called.

### CODE REVIEW

File: lib/application.js [hunk 1]
```diff
@@ -99,7 +99,7 @@ module.exports = class Application extends Emitter {
    */
 
   use(fn) {
-    debug('use %s', fn._name || fn.name || '-');
+    debug('use %s', fn && (fn._name || fn.name) || '-');
```
Comment (dead-horse) [inline lib/application.js]: just move this line below `if (isGeneratorFunction(fn)) ...` ?

Comment (jonathanong) [inline lib/application.js]: yeah i prefer this solution

Comment (yanickrochon) [inline lib/application.js]: I had an initial `if (fn === null || fn === undefined)`, but decided that it wasn't necessary... I also debated moving `debug` below the `if` (for type and generator check), but figured that having a debug output before throwing could be useful :)

**Comment by dead-horse** [comment]
please change the target branch to `v2.x`.

**Comment by yanickrochon** [comment]
Holy crap! This wouldn't be needed had it been done before :P

**Pull request closed without merging**